backpack/custom_module/graph_utils.py

```diff
@@ -196,11 +238,161 @@ def _transform_flatten_to_module(module: Module, debug: bool) -> GraphModule:
+    graph: Graph = BackpackTracer().trace(module)
+
+    nodes = [
+        n
+        for n in graph.nodes
+        if n.op == "call_function" and target_function in str(n.target)
+    ]
+    nodes += [
+        n
+        for n in graph.nodes
+        if n.op == "call_method" and target_method == str(n.target)
+    ]
+
+    for node in nodes:
+        print(node.args)
+        _change_node_to_module(
+            node,
+            "permute",
+            module,
+            Permute(*node.args[1:], init_transpose=True),
+            (node.args[0],),
+        )
+        counter += 1
+
+    graph.lint()
+    if debug:
+        print(f"\tPermute transformed: {counter}")
+    return GraphModule(module, graph)
+
+
+def _transform_lstm_rnn(module: Module, debug: bool) -> GraphModule:
+    if debug:
+        print("\tBegin transformation: LSTM, RNN")
+    counter: int = 0
+    graph: Graph = BackpackTracer().trace(module)
+
+    for node in graph.nodes:
+        if node.op == "call_module" and isinstance(
+            module.get_submodule(node.target), (RNN, LSTM)
+        ):
+            lstm_module: Union[RNN, LSTM] = module.get_submodule(node.target)
+            if lstm_module.num_layers > 1:
+                if len(node.args) > 1:
+                    raise NotImplementedError(
+                        "For conversion, LSTM input must not have hidden states."
+                    )
+                lstm_module_replace = _make_rnn_backpack(lstm_module)
+                module.add_module(node.target, lstm_module_replace)
+                counter += 1
+
+    graph.lint()
+    if debug:
+        print(f"\tRNNs, LSTMs transformed: {counter}")
+    return GraphModule(module, graph)
+
+
+def _rnn_hyperparams(module: Union[RNN, LSTM]) -> Tuple[int, int, float, bool]:
+    if module.bias is not True:
+        raise NotImplementedError("only bias = True is supported")
+    if module.bidirectional is not False:
+        raise NotImplementedError("only bidirectional = False is supported")
+    if isinstance(module, RNN):
+        if module.nonlinearity != "tanh":
+            raise NotImplementedError("only nonlinearity = 'tanh' is supported")
+    elif isinstance(module, LSTM):
+        if module.proj_size != 0:
+            raise NotImplementedError("only proj_size = 0 is supported")
+    return module.input_size, module.hidden_size, module.dropout, module.batch_first
+
+
+def _make_rnn_backpack(module: Union[RNN, LSTM]) -> Module:
+    input_size, hidden_size, dropout, batch_first = _rnn_hyperparams(module)
+    rnn_class = type(module)
+    rnn_module_replace = Sequential()
+    for layer in range(module.num_layers):
+        rnn_layer = rnn_class(
+            input_size if layer == 0 else hidden_size,
+            hidden_size,
+            batch_first=batch_first,
+        )
+        for param_str in ["weight_ih_l", "weight_hh_l", "bias_ih_l", "bias_hh_l"]:
+            setattr(rnn_layer, f"{param_str}0", getattr(module, f"{param_str}{layer}"))
```

Comment:
Okay, it could be that `register_parameter` creates a new parameter name if it detects that the name you are trying to assign a parameter to is already taken. Keep the old version.